### DISCUSSION TITLE
Websocket Message Parser

### DISCUSSION THREAD

**Discussion by astrebaba31** [synfron/ReshaperForBurp#54] [Ideas]
For example, let's say we receive a message like 42(5[1,2,3,4,5]) from a WebSocket. I want to extract the value 5 from within this message. It would be helpful if such a feature were available.

**Comment by ddwightx** [comment]
Hi @astrebaba31, you can do that using Set Variable under WebSocket Rules.
* Then
  * Set Variable
    * Use Message Value - Checked
    * Source Message Value - WebSocket Message
    * Source Value Type - Text
    * Use Regex Replace - Checked
    * Regex Pattern - `\d+\((\d+)\[(\d+,?)*\]\)`
    * Regex Replacement Text - `$1`
    * Destination Variable Source - `<Your choice>`
    * Destination Variable Name - `<Your choice>`

`WebSocket Message` gets you the value `42(5[1,2,3,4,5])` and the regex extracts the subvalue.
